packages/comet-extras/src/components/data-table/data-table.tsx

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+import {
+  SortingState,
+  flexRender,
+  getCoreRowModel,
+  getSortedRowModel,
+  useReactTable,
+  getPaginationRowModel,
+  PaginationState,
+} from '@tanstack/react-table';
+import classNames from 'classnames';
+import './data-table.style.css';
+
+export interface DataTableProps<T = any> {
+  /**
+   * The unique identifier for this component
+   */
+  id: string;
+  /**
+   * The table column definitions
+   */
+  columns: T[];
+  /**
+   * The data to display in the table rows
+   */
+  data: T[];
+  /**
+   * A boolean indicating if the table is striped or not
+   */
+  striped?: boolean;
+  /**
+   * A boolean indicating if the table is sortable or not
+   */
+  sortable?: boolean;
+  /**
+   * The column id to set as the default sort
+   */
+  sortCol?: string;
+  /**
+   * The default sort direction if sortIndex is provided
+   */
+  sortDir?: 'asc' | 'desc';
+  /**
+   * A boolean indicating if the table is pageable or not
+   */
+  pageable?: boolean;
+  /**
+   * The index of the page to display as default
+   */
+  pageIndex?: number;
+  /**
+   * The number of items to display per page
+   */
+  pageSize?: number;
+  /**
+   * Additional class names for the table
+   */
+  className?: string;
+}
+/**
+ * A Data Table shows information in columns and rows, with advanced functionality.
+ */
+export const DataTable = ({
+  id,
+  columns,
+  data,
+  striped = false,
+  sortable = false,
+  sortCol,
+  sortDir,
+  pageable = false,
+  pageIndex = 0,
+  pageSize = 10,
+  className,
+}: DataTableProps): React.ReactElement => {
+  const [sorting, setSorting] = React.useState<SortingState>(
+    sortable ? [{ id: sortCol ?? columns[0], desc: sortDir === 'desc' }] : [],
+  );
+  const [paging, setPaging] = React.useState<PaginationState>({ pageIndex, pageSize });
+  const table = useReactTable({
+    data,
+    columns,
+    state: {
+      sorting,
+      pagination: paging,
+    },
+    enableSorting: sortable,
+    onSortingChange: setSorting,
+    onPaginationChange: setPaging,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+  });
+
+  return (
+    <>
+      <table id={id} className={classNames('data-table', { striped }, className)}>
+        <thead>
+          {table.getHeaderGroups().map((headerGroup) => (
+            <tr key={headerGroup.id}>
+              {headerGroup.headers.map((header) => (
+                <th key={header.id}>
+                  <div
+                    {...{
```

Comment:
Interesting choice. You're creating an object just to destructure it. Why not just keep it simple?

```tsx
<div
  className={header.column.getCanSort() ? 'cursor-pointer select-none' : ''}
  onClick={header.column.getToggleSortingHandler()}
>
```

Reply:
Good catch. I believe this was left over from when this section had additional configurations. Removed